deps/rabbitmq_stomp/src/rabbit_stomp_processor.erl

```diff
@@ -684,117 +690,383 @@ do_subscribe(Destination, DestHdr, Frame,
 
             Props = rabbit_stomp_util:message_properties(Frame1),
 
-            {Exchange, RoutingKey} = parse_routing(Destination, DfltTopicEx),
-
-            Method = #'basic.publish'{
-              exchange = list_to_binary(Exchange),
-              routing_key = list_to_binary(RoutingKey),
-              mandatory = false,
-              immediate = false},
-
-            case transactional(Frame1) of
-                {yes, Transaction} ->
-                    extend_transaction(
-                      Transaction,
-                      fun(StateN) ->
-                              maybe_record_receipt(Frame1, StateN)
-                      end,
-                      {Method, Props, BodyFragments},
-                      State1);
-                no ->
-                    ok(send_method(Method, Props, BodyFragments,
-                                   maybe_record_receipt(Frame1, State1)))
-            end;
+            {ExchangeNameList, RoutingKeyList} = parse_routing(Destination, DfltTopicEx),
+            %% io:format("Parse_routing: ~p~n", [{ExchangeNameList, RoutingKeyList}]),
+            RoutingKey = list_to_binary(RoutingKeyList),
+
+
+            rabbit_global_counters:messages_received(stomp, 1),
+
+            ExchangeName = rabbit_misc:r(VHost, exchange, list_to_binary(ExchangeNameList)),
+            check_resource_access(User, ExchangeName, write, AuthzCtx),
+            Exchange = rabbit_exchange:lookup_or_die(ExchangeName),
+            check_internal_exchange(Exchange),
+            check_topic_authorisation(Exchange, User, RoutingKey, AuthzCtx, write),
+
+            {DeliveryOptions, _MsgSeqNo, State2} =
+                case rabbit_stomp_frame:header(Frame, ?HEADER_RECEIPT) of
+                    not_found ->
+                        {maps_put_truthy(flow, Flow, #{}), undefined, State1};
+                    {ok, Id} ->
+                        rabbit_global_counters:messages_received_confirm(stomp, 1),
+                        SeqNo = State1#proc_state.msg_seq_no,
+                        %% I think it's safe to just add it here because
+                        %% if there is an error down the road process dies
+                        StateRR = record_receipt(true, SeqNo, Id, State1),
+                        Opts = maps_put_truthy(flow, Flow, #{correlation => SeqNo}),
+                        {Opts, SeqNo, StateRR#proc_state{msg_seq_no = SeqNo + 1}}
+                end,
 
-        {error, _} = Err ->
+            {ClassId, _MethodId} = rabbit_framing_amqp_0_9_1:method_id('basic.publish'),
+
+            Content0 = #content{
+                          class_id = ClassId,
+                          properties = Props,
+                          properties_bin = none,
+                          protocol = none,
+                          payload_fragments_rev = [BodyFragments]
+                         },
 
+            Message0 = mc_amqpl:message(ExchangeName, RoutingKey, Content0),
```

Comment:
I think it would be great to directly store STOMP messages as part of this PR and to implement a `mc_stomp` module that does the translations. It's unnecessary to convert from / to AMQP 0.9.1 message format. But it can also be done as a separate PR.